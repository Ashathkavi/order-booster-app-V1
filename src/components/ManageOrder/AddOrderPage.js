import React from 'react'
import {OrderForm} from './OrderForm'
import {addOrder} from '../../actions/orders'
import {connect} from 'react-redux'


export const AddOrderPage = (props) => {

    const onSubmit = (order) => {
        props.addOrder(order)
        props.history.push('/order')
    }

    return(
        <div>
            <h2>Add Order</h2>
            <OrderForm onSubmit={ onSubmit}/>

        </div>
)}


const mapDispatchToProps = (dispatch) => ({
    addOrder: (order) => dispatch(addOrder(order))
})

export default connect(undefined, mapDispatchToProps)(AddOrderPage)