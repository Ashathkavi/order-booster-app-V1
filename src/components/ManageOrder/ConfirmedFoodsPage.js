import React, {useEffect} from 'react'
import {connect} from 'react-redux'
import {startEditOrder} from '../../actions/orders'
import selectedOrders from '../../selectors/orders'
import {setStatusFilter} from '../../actions/orderFilters'
import categoryList from '../../fixtures/FoodCategoryList'
import moment from 'moment'


export const ConfirmedFoodsPage = (props) => {
    console.log(props.orders, 'props.orders')


    let soups=[] , apetizer =[], chopsueys=[], riceAndCurry=[], friedRice=[], briyani=[]
    let noodles=[] , spaghetti =[], pasta=[], kothu=[], vegetableDishes=[], eggDishes=[]
    let fishDishes=[] , chickenDishes =[], prawnDishes=[], cuttleFishDishes=[], crabDishes=[], desserts=[]
    let milkShakes=[] , freshFruitJuice =[], softDrink=[], sandwiches=[], specialMenue=[]
    
    props.orders.map((order)=>order.foods.map((food)=>{
        //console.log(food.food.category, 'food.food.category')
        if(food.food.category === 'soups'){
            soups.push({...food, orderNo:order.count, index:order.foods.indexOf(food), orderId:order.id})
        }else if (food.food.category === 'Appetizer'){
            apetizer.push({...food, orderNo:order.count, index:order.foods.indexOf(food), orderId:order.id})
        }else if (food.food.category === 'Chopsueys'){
            chopsueys.push({...food, orderNo:order.count, index:order.foods.indexOf(food), orderId:order.id})
        }else if (food.food.category === 'Rice and Curry'){
            riceAndCurry.push({...food, orderNo:order.count, index:order.foods.indexOf(food), orderId:order.id})
        }else if (food.food.category === 'Fried Rice'){
            friedRice.push({...food, orderNo:order.count, index:order.foods.indexOf(food), orderId:order.id})
        }else if (food.food.category === 'Briyani'){
            briyani.push({...food, orderNo:order.count, index:order.foods.indexOf(food), orderId:order.id})
        }else if (food.food.category === 'Noodles'){
            noodles.push({...food, orderNo:order.count, index:order.foods.indexOf(food), orderId:order.id})
        }else if (food.food.category === 'Spaghetti'){
            spaghetti.push({...food, orderNo:order.count, index:order.foods.indexOf(food), orderId:order.id})
        }else if (food.food.category === 'Pasta'){
            pasta.push({...food, orderNo:order.count, index:order.foods.indexOf(food), orderId:order.id})
        }else if (food.food.category === 'Kothu'){
            kothu.push({...food, orderNo:order.count, index:order.foods.indexOf(food), orderId:order.id})
        }else if (food.food.category === 'Vegetable Dishes'){
            vegetableDishes.push({...food, orderNo:order.count, index:order.foods.indexOf(food), orderId:order.id})
        }else if (food.food.category === 'Egg Dishes'){
            eggDishes.push({...food, orderNo:order.count, index:order.foods.indexOf(food), orderId:order.id})
        }else if (food.food.category === 'Fish Dishes'){
            fishDishes.push({...food, orderNo:order.count, index:order.foods.indexOf(food), orderId:order.id})
        }else if (food.food.category === 'Chicken Dishes'){
            chickenDishes.push({...food, orderNo:order.count, index:order.foods.indexOf(food), orderId:order.id})
        }else if (food.food.category === 'Prawn Dishes'){
            prawnDishes.push({...food, orderNo:order.count, index:order.foods.indexOf(food), orderId:order.id})
        }else if (food.food.category === 'CuttleFish Dishes'){
            cuttleFishDishes.push({...food, orderNo:order.count, index:order.foods.indexOf(food), orderId:order.id})
        }else if (food.food.category === 'Crab Dishes'){
            crabDishes.push({...food, orderNo:order.count, index:order.foods.indexOf(food), orderId:order.id})
        }else if (food.food.category === 'Desserts'){
            desserts.push({...food, orderNo:order.count, index:order.foods.indexOf(food), orderId:order.id})
        }else if (food.food.category === 'MilkShakes'){
            milkShakes.push({...food, orderNo:order.count, index:order.foods.indexOf(food), orderId:order.id})
        }else if (food.food.category === 'Fresh Fruit Juices'){
            freshFruitJuice.push({...food, orderNo:order.count, index:order.foods.indexOf(food), orderId:order.id})
        }else if (food.food.category === 'Soft Drink'){
            softDrink.push({...food, orderNo:order.count, index:order.foods.indexOf(food), orderId:order.id})
        }else if (food.food.category === 'Sandwiches'){
            sandwiches.push({...food, orderNo:order.count, index:order.foods.indexOf(food), orderId:order.id})
        }else if (food.food.category === 'Special Menue'){
            specialMenue.push({...food, orderNo:order.count, index:order.foods.indexOf(food), orderId:order.id})
        }
    }))

    console.log(spaghetti, 'spaghetti')

    const categoryList = [
        soups , apetizer , chopsueys, riceAndCurry, friedRice, briyani,
        noodles , spaghetti , pasta, kothu, vegetableDishes, eggDishes,
        fishDishes , chickenDishes , prawnDishes, cuttleFishDishes, crabDishes, desserts,
        milkShakes , freshFruitJuice , softDrink, sandwiches, specialMenue
    ]

    const onChangePreparedStatus = (id, food, orderedFoods) => {
        let foods = orderedFoods
        foods[food.index].prepared = true
        foods[food.index].preparedBy = props.auth.uid
        foods[food.index].preparedTime = moment().valueOf()
        let orderReady = true
        foods.map((food)=> {
            if(food.prepared === false){
                orderReady = false
            }
        })
        if(orderReady){
            props.startEditOrder(id ,{foods:foods, status:{status:'table', time:moment().valueOf()}})
        }else{
            props.startEditOrder(id ,{foods:foods})
        }
        console.log(foods, 'Foods', id)
        
    }


    useEffect(()=>{
        if(props.filters.status !== 'kitchen'){
            props.setStatusFilter('kitchen')
        }
    }, ()=>{

    })


    
    return(
        <div>
        {
            categoryList.map((category)=>{
                
                if(category.length !== 0){
                    console.log(category, 'category')
                    return (
                        <div>
                            <h3>{category[0].food.category}</h3>
                            {
                                
                                category.map((food)=>{
                                    let orderedFoods = []
                                    props.orders.map((order)=>{
                                        if(order.id === food.orderId){
                                            orderedFoods = order.foods
                                            console.log(orderedFoods, 'orderedFoods')
                                        }
                                    })
                                    
                                    return (
                                        <div>
                                            <p>{food.food.name}:::
                                            {food.food.foodSize}:::
                                            {food.foodQuantity}:::
                                            {food.orderNo}:::</p>
                                            {food.prepared === false && <button onClick={()=>onChangePreparedStatus(food.orderId, food, orderedFoods)}>Mark it as Prepared</button>}

                                        </div>
                                    )
                                        
                                })
                            }
                        </div>
                    )
                }
            })
        }
        </div>
        
            
        
    )
}


const mapDispatchToProps = (dispatch) => ({
    startEditOrder: (id, order) => dispatch(startEditOrder(id, order)),
    setStatusFilter: (status) => dispatch(setStatusFilter(status))

})

const mapStateToProps = (state) => {
    return {        
        orders:selectedOrders(state.orders, state.orderFilters),
        filters:state.orderFilters,
        auth:state.auth
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(ConfirmedFoodsPage)