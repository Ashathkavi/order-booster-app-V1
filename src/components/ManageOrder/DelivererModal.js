import React, {useEffect, useState} from 'react'
import Modal from 'react-modal'
import CategoryList from '../../fixtures/FoodCategoryList'
import database from '../../firebase/firebase'


class Deliverer{
    constructor(name, uid){
        this.name = name
        this.uid = uid
    }
}

const useFetchDeliverer = () => {
    const [value, setValue] = useState([]);
    useEffect(()=>{
        const deliverers = []
        database.ref(`users`).once('value')
            .then((snap)=>{
                snap.forEach((childSnapshot)=>{
                    if(childSnapshot.val().role === 'deliverer'){
                        const deliverer = new Deliverer(childSnapshot.val().name, childSnapshot.key)
                        deliverers.push(deliverer)                        
                    }
                })
                setValue(deliverers)                    
            })
            .catch((error)=>console.log('User gaining request failed :', error))
        
    },[])
    return [value, setValue]
}



const DelivererModal = (props) => {

    const [deliverers, setDeliverers] = useFetchDeliverer()

    

    const arrayBuilder = (array, devideBy) => {
        let DelivererListV1 = []
        let DelivererListV2 = []
        let i = 0
        array.map((item)=>{
            DelivererListV1.push(item)
            i = i +1

            if(i===devideBy){
                DelivererListV2.push(DelivererListV1)
                i=0
                DelivererListV1=[]
            }

            if(array.indexOf(item) === array.length - 1){
                DelivererListV2.push(DelivererListV1)
                console.log()
            }

        }) 
        return DelivererListV2

    }
    const modifiedDelList = arrayBuilder(deliverers, 2)
    

    return(
        <Modal
            isOpen={props.modalIsOpen}
            contentLabel = "Please Select a Deliverer"
            onRequestClose = {props.handleCloseModal}
        >
            <h3 className="modal__header" >Please Select a Deliverer</h3>
            <div>
            
                {
                    modifiedDelList.map((internalDelList)=>{
                        console.log(modifiedDelList)
                        return(
                            <div className="modal__content">
                                {
                                    internalDelList.map((deliverer)=> (
                                        <div className="modal__content__item__deliverer">
                                            <button className="button button--selectFood" onClick={() =>{                
                                                props.onDelivery(deliverer.uid)
                                                props.handleCloseModal()
                                            }}>{deliverer.name}</button>
                                        </div>
                                    ))
                                    
                                }
                                {
                                    internalDelList[1] === undefined ? <div className="modal__content__item__deliverer"></div> : null
                                    
                                }
                            </div>
                        )
                    })
                }
            </div>
            <div className="modal__buttons">
                    
                <button className="button button--closeModal" onClick={()=>props.handleCloseModal()}>Close</button>
                <button className="button button--selectNone" onClick={()=>{
                    props.onDelivery('') 
                    props.handleCloseModal()
                }}>Select None</button>
            </div>
            
        </Modal>
)}

export default DelivererModal;


/*

*/