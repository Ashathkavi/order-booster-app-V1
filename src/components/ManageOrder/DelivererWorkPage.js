import React, {useEffect, useState} from 'react'
import { connect } from 'react-redux'
import moment from 'moment'
import numeral from 'numeral'
import {startEditOrder} from '../../actions/orders'
import {setDeliverMethFilter, setStartDate} from '../../actions/orderFilters'
import OrderViewModal from './OrderViewModal'
import {onStartSetOrders} from '../../actions/orders'
import selectedOrders from '../../selectors/orders'



export const DelivererWorkPage = ({
    orders,
    setDeliverMethFilter,
    onStartSetOrders,
    setStartDate,
    startEditOrder

    }) => {
    //console.log(props)

    useEffect(()=>{
        setDeliverMethFilter('delivery')
        setStartDate(moment().subtract(1,'month').startOf('month').valueOf())
        const g = onStartSetOrders()
        return g
    },[])
    const [orderViewIsOpen, setOrderViewIsOpen ] = useState(false)

    const now = moment()

    const onChangeStatus = () => {
        startEditOrder(id ,{status:{status:'recieved', time:now.valueOf()}})
    }


    return(
        <div className='content-container'>
        
            <div className="list-body--deliverer">
            {
                orders.map((order)=>(
                    <div className="list-item list-item--deliverer">
                        
                        <h2 className="list-item__title list-item__title--deliverer">{order.count}  &nbsp;&nbsp;&nbsp; &nbsp;&nbsp;&nbsp; {numeral(order.amount).format('$0,0.00')}</h2>
                        <p><b>Customer Name: </b> {order.customerName}</p>
                        <p><b>Phone Number: </b>{order.phoneNumber}</p>
                        <p><b>Address: </b>{order.address}</p>    
                        {order.status.status === 'on delivery' && <button onClick={onChangeStatus}>Mark it as Delivered</button>}
                        <p >{moment(order.orderEndTime).format('MMMM Do, YYYY   h:mm a') }</p>
                        <button className="button button--buttonMainBoard" onClick={()=>setOrderViewIsOpen(true)}>See Order Details</button> &nbsp;&nbsp;&nbsp;
                    
                        <OrderViewModal 
                            modalIsOpen={orderViewIsOpen} 
                            handleCloseModal={()=>setOrderViewIsOpen(false)} 
                            {...order}                
                            startEditOrder={startEditOrder}
                            from={'delivererList'}
                        />
                                
                    </div>
                ))
            }
            </div>
        </div>
        
    )

}



const mapDispatchToProps = (dispatch) => ({
    startEditOrder: (id, order) => dispatch(startEditOrder(id, order)),
    setDeliverMethFilter:(deliverMeth) => dispatch(setDeliverMethFilter(deliverMeth)),
    setStartDate:(startDate)=>dispatch(setStartDate(startDate)),
    onStartSetOrders: () => dispatch(onStartSetOrders())

})

const mapStateToProps = (state) => {
    return {        
        orders:selectedOrders(state.orders, state.orderFilters),
        filters:state.orderFilters,
        auth:state.auth
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(DelivererWorkPage)