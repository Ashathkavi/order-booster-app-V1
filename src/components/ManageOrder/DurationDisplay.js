import React, {useState,useEffect} from 'react'
import moment from 'moment'


const DurationDisplay = ({orderEndTime}) => {
    const [time, setTime] = useState(moment().valueOf())

    useEffect(() => {
        setTimeout(() => setTime(moment().valueOf()), 60000)
    }, [time])

    let duration = moment(orderEndTime).diff(moment())
    let durationPlusTen = moment(orderEndTime).subtract(10,'minutes').diff(moment())
    if(duration < 0){
        if(moment.utc(duration*-1).format('DD')==='01'){
            return <span style={{color: "red"}}>{moment.utc(duration*-1).format('HH ')}h&nbsp;&nbsp;{moment.utc(duration*-1).format('mm ')}min</span>
        }else{
            return <span style={{color: "red"}}>
                        {moment.utc(duration*-1).format('DD ')}d&nbsp;&nbsp;
                        {moment.utc(duration*-1).format('HH ')}h&nbsp;&nbsp;
                        {moment.utc(duration*-1).format('mm ')}min
                    </span>
        }
        //console.log(moment.utc(duration*-1).format('DD'))
    }else if(durationPlusTen < 0){
        return <span style={{color: "orange"}}>{moment.utc(duration).format('mm ') }min</span>
    }else{
        if(moment.utc(duration).format('HH')==='00'){
            return <span >{moment.utc(duration).format('mm ')}min</span>
        }else{
            return <span >{moment.utc(duration).format('HH ')}h&nbsp;&nbsp;{moment.utc(duration).format('mm ')}min</span>
        }
        
    }
    

} 

export default DurationDisplay