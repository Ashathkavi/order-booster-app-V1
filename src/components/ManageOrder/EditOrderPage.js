import React, {useState} from 'react'
import {connect} from 'react-redux'
import { startEditOrder, startRemoveOrder} from '../../actions/orders'
import OrderForm from './OrderForm'
import MessageModal from '../MessageModal'
import selectFood from '../../selectors/foods'


export const EditOrderPage = (props) => {
    //console.log(props, 'props')

    const [visibleMessageModal, setVisibleMessageModal] = useState(false)
    
    const onSubmit = (order) => {
        console.log(props.order, 'props.order')
        props.startEditOrder(props.order.id ,order, props.order)
        props.history.push('/order')
    }

    const toggleVisibleModal = () =>{
        setVisibleMessageModal(!visibleMessageModal)
    }

    const onRemove = () => {
        props.startRemoveOrder(props.order.id)
        props.history.push('/order')
    }

    return(
        <div>
            <div className="page-header">
                <div className="content-container">
                    <h1 className="page-header__title">Edit Order</h1>
                </div>
            </div>
           <div className="content-container">
                <OrderForm order={props.order} onSubmit={onSubmit} foods={props.foods}/>
                {props.autherizedAs === 'admin' && <button className="button button--remove" onClick={toggleVisibleModal}>Remove this Order</button>}

           </div>
           <MessageModal 
                onVisibleMessageModal={toggleVisibleModal} 
                isMessageModalOpen = {visibleMessageModal}
                onSubmit={onRemove}
                orderNo={props.order.count}
                message='Are You sure, Do you want to delete the Order'
                title='Order Delete Alert'
           />
        </div>
    )
}


const mapDispatchToProps = (dispatch) => ({
    startEditOrder: (id, order, prev) => dispatch(startEditOrder(id, order, prev)),
    startRemoveOrder: (id) => dispatch(startRemoveOrder({id}))
})

const mapStateToProps = (state, props) => {
    //console.log(state.orders)
    return {
        
        order:state.orders.find((order)=> order.id === props.match.params.id),
        autherizedAs:state.auth.role,
        foods:selectFood(state.foods, state.foodFilters)
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(EditOrderPage)