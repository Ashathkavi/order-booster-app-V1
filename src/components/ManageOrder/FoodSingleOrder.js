import React from 'react'
import {Link} from 'react-router-dom'

const FoodSingleOrder = (props) => (
    <div>
        <p>
            <span>{props.singleTypeOrder.food.name}</span>::::::::
            <span>{props.singleTypeOrder.foodQuantity}</span>:::::
            <span>{props.iterable}</span>
            <button type='button' onClick={()=>props.onRemoveSingleOrder(props.iterable)}>Remove</button>
        </p>
    </div>
)

export default FoodSingleOrder