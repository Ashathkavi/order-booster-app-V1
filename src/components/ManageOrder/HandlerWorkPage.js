import React, {useEffect, useState} from 'react'
import { connect } from 'react-redux'
import moment from 'moment'
import {startEditOrder} from '../../actions/orders'
import {setStatusFilter, setStartDate} from '../../actions/orderFilters'
import {onStartSetOrders} from '../../actions/orders'
import selectedOrders from '../../selectors/orders'
import DurationDisplay from './DurationDisplay'
import MessageModal from '../MessageModal'
import {GiStabbedNote} from "react-icons/gi";





export const HandlerWorkPage = ({
    orders,
    setStatusFilter,
    onStartSetOrders,
    setStartDate,
    startEditOrder,
    auth

    }) => {
    //console.log(props)

    const [isFoodDisModalOpen, setIsFoodDisModalOpen] = useState(false)
    const [foodDescription, setFoodDescription] = useState('')

    
    const onFoodDescription = (foodDescription) => {
        setIsFoodDisModalOpen(!isFoodDisModalOpen)
        setFoodDescription(foodDescription)    
    }

    useEffect(()=>{
        setStatusFilter('kitchen')
        setStartDate(moment().subtract(1,'month').startOf('month').valueOf())
        const g = onStartSetOrders()
        return g
    },[])
    const onChangeHandledStatus = (id, foodIndex, orderedFoods) => {
        let foods = orderedFoods
        foods[foodIndex].handled = true
        foods[foodIndex].handledBy = auth.uid
        foods[foodIndex].handledTime = moment().valueOf()
        let orderReady = true
        foods.map((food)=> {
            if(food.handled === false){
                orderReady = false
            }
        })
        if(orderReady){
            startEditOrder(id ,{foods:foods, status:{status:'table', time:moment().valueOf()}})
            console.log('orderready')
        }else{
            startEditOrder(id ,{foods:foods})
        }
        //console.log(foods, 'Foods', id)
        
    }
    
    return(
        <div className='content-container'>
        
            <div className="list-body--deliverer">
            {
                orders.map((order)=>(
                    <div className="list-item list-item--deliverer">
                
                        <h2 className="list-item__title list-item__title--deliverer">{order.count}  &nbsp;&nbsp;&nbsp; &nbsp;&nbsp;&nbsp; {order.deliverMeth}</h2>
                        <p><b>Order status: </b> {order.status.status}</p>
                        <table className="table-list-item-handler">
                            {
                                order.foods.map((food)=>{
                                    return (
                                        <tr>
                                            <td>{food.food.name}{!!food.foodDescription && <GiStabbedNote onClick={()=>onFoodDescription(food.foodDescription)}/>} </td>
                                            <td>{food.food.foodSize}</td>
                                            <td>{food.foodQuantity}</td>
                                            <td>{!!food.prepared ? 'Prepared' : 'Not Prepared'}</td>
                                            { (!food.handled && !!food.prepared ) &&<td> <button onClick={()=>onChangeHandledStatus(order.id, order.foods.indexOf(food), order.foods)}>Mark as Handled</button></td>}
                                            { (!food.handled && !food.prepared ) &&<td> Not Handled</td>}
                                            { (!!food.handled && !!food.prepared ) &&<td>  Handled</td>}

                                        </tr>)
                                })
                            }
                        </table>
                        
                        
                        {!!order.description && <p><b>Description: </b>{order.description}</p>  }  
                        <div className="list-item__duration"><DurationDisplay orderEndTime={order.orderEndTime}/></div>
                    </div>
                ))
            }
            </div>
            <MessageModal
                onVisibleMessageModal={setIsFoodDisModalOpen} 
                isMessageModalOpen = {isFoodDisModalOpen}
                message={foodDescription}
                title={''}
            />
        </div>
    )

}



const mapDispatchToProps = (dispatch) => ({
    startEditOrder: (id, order) => dispatch(startEditOrder(id, order)),
    setStatusFilter:(status) => dispatch(setStatusFilter(status)),
    setStartDate:(startDate)=>dispatch(setStartDate(startDate)),
    onStartSetOrders: () => dispatch(onStartSetOrders())

})

const mapStateToProps = (state) => {
    return {        
        orders:selectedOrders(state.orders, state.orderFilters),
        filters:state.orderFilters,
        auth:state.auth
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(HandlerWorkPage)