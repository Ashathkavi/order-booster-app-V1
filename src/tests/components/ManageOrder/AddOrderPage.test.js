import React from 'react'
import {shallow} from 'enzyme'
import {AddOrderPage} from '../../../components/ManageOrder/AddOrderPage'
import sampleOrders from '../../../fixtures/sampleOrders'

const sample_orders = sampleOrders()

let startAddOrderSpy, historySpy, wrapper
 
beforeEach(()=>{
    startAddOrderSpy = jest.fn()
    historySpy = {push: jest.fn()}
    wrapper = shallow(<AddOrderPage startAddOrder={startAddOrderSpy} history={historySpy}/>)
})

test('should render AddOrderPage correctly', () => {
    expect(wrapper).toMatchSnapshot();
})

test('should handle onSubmit', () => {
    wrapper.find('OrderForm').prop('onSubmit')(sample_orders[0])
    expect(historySpy.push).toHaveBeenLastCalledWith('/order')
    expect(startAddOrderSpy).toHaveBeenLastCalledWith(sample_orders[0])
})