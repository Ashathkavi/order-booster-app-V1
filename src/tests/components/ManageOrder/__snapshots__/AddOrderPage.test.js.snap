// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AddOrderPage correctly 1`] = `
<div>
  <h2>
    Add Order
  </h2>
  <OrderForm
    onSubmit={[Function]}
  />
</div>
`;
